californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -137,6 +144,12 @@
 	/** The matcher which matches incoming responses, akcs and rsts an exchange */
 	private final Matcher matcher;
 
+	/** Serializer to convert messages to datagrams. */
+	private final Serializer serializer;
```

Comment:
Can we make UdpSerializer and TcpSerializer static utility classes? Or do they need to keep some state?